src/lib/Embed.hs

```diff
@@ -388,13 +408,17 @@ instance MonadError e m => MonadError e (EmbedT m) where
 
 getNameHint :: MonadEmbed m => m Name
 getNameHint = do
-  v <- fst <$> embedAsk
-  return $ case v of
-    NoName -> "tmp"
-    _      -> v
+  tag <- fst <$> embedAsk
+  return $ Name GenName tag 0
 
+-- This is purely for human readability. `const id` would be a valid implementation.
```

Comment:
Yes, good point. Replacing with `const id` might break things for that reason. At some point I'd like to be able to run our test under an "ignore hints" flag so that we can check that we're not unintentionally relying on them.